README.md

```diff
@@ -4,10 +4,84 @@ This repository contains a Javascript implementation of [esptool](https://github
 
 `esptool-js` is based on [Web Serial API](https://wicg.github.io/serial/) and works in Google Chrome and Microsoft Edge, [version 89 or later](https://developer.mozilla.org/en-US/docs/Web/API/Serial#browser_compatibility).
 
+
+## Usage
+
+CDN
+
+``
+
+or import as npm package
+
+`npm install --save esptool-js`
+
+`yarn add --save esptool-js`
+
+Example code:
+
+```ts
+import { ESPLoader, Transport } from "esptool-js";
+
+const portFilters : { usbVendorId?: number | undefined; usbProductId?: number | undefined;}[] = [];
+const device = await navigator.serial.requestPort({ filters: portFilters });
+const transport = new Transport(device);
+
+// You can use any JavaScript compatible terminal by wrapping a helper object like this:
+let espLoaderTerminal = {
+  clean() {
+    // Implement the clean function for your terminal here.
+  },
+  writeLine(data) {
+    // Implement the writeLine function for your terminal here.
+  },
+  write(data) {
+    // Implement the write function for your terminal here.
+  },
+};
+
+const esploader: ESPLoader = new ESPLoader(transport, baudRateInteger, espLoaderTerminal);
```

Comment:
If the terminal is not set, it will print to the console. That will be a better experience than a fake terminal. I would keep the example focused on the user getting something to work, not learn about all the options at the same time.
```suggestion
const esploader: ESPLoader = new ESPLoader(transport, baudRateInteger);
```